packages/server/src/db.ts

```diff
@@ -53,11 +55,27 @@ export function makeKnexConfig(dbConfig: DatabaseConfig): KnexDatabaseConfig {
 	if (dbConfig.client === 'sqlite3') {
 		connection.filename = dbConfig.name;
 	} else {
-		connection.database = dbConfig.name;
-		connection.host = dbConfig.host;
-		connection.port = dbConfig.port;
-		connection.user = dbConfig.user;
-		connection.password = dbConfig.password;
+		if(dbConfig.connectionString){
+			connection.connectionString = dbConfig.connectionString;
+		} else {
+			connection.database = dbConfig.name;
+			connection.host = dbConfig.host;
+			connection.port = dbConfig.port;
+			connection.user = dbConfig.user;
+			connection.password = dbConfig.password;
+		}
+
+		connection.ssl = {};
+		if (dbConfig.sslCaFilePath)
+			connection.ssl.ca = fs.readFileSync(dbConfig.sslCaFilePath);
+
+		if (dbConfig.sslCertFilePath && dbConfig.sslCertKeyFilePath) {
+			connection.ssl.cert = fs.readFileSync(dbConfig.sslCertFilePath);
+			connection.ssl.key = fs.readFileSync(dbConfig.sslCertKeyFilePath);
+		}
```

Comment:
I feel these files should be loaded earlier than that, and it's also not ideal to call sync methods here as it can make the tests that run in parallel slower.

Instead, could you change it like so please?

- Rename sslCaFilePath, sslCertFilePath and sslCertKeyFilePath to just sslCa, sslCert and sslKey, which will contain the already loaded data (that will also allow setting these values from something other than files).

- Make `databaseConfigFromEnv` async, and load the files in this function, using `await fs.readFile()`

Reply:
will do